File: project_flutter/lib/models/projet.dart
class Projet {
  final int id;
  final String nom;
  final String client;
  final String? description;
  final String dateDebut;
  final String dateFin;
  final String statut;
  final String priorite;

  Projet({
    required this.id,
    required this.nom,
    required this.client,
    this.description,
    required this.dateDebut,
    required this.dateFin,
    required this.statut,
    required this.priorite,
  });

  factory Projet.fromJson(Map<String, dynamic> json) {
    return Projet(
      id: json['id'],
      nom: json['nom'],
      client: json['client'],
      description: json['description'],
      dateDebut: json['date_debut'],
      dateFin: json['date_fin'],
      statut: json['statut'],
      priorite: json['priorite'],
    );
  }
}

File: backend_laravel/app/Models/Projet.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use App\Models\User;
use App\Models\Tache;
use App\Models\Rapport;

class Projet extends Model
{
    protected $fillable = [
        'nom',
        'description',
        'client',
        'client_id',
        'chef_projet_id',
        'date_debut',
        'date_fin',
        'statut',
        'priorite',
    ];

    // Un projet appartient à un chef de projet (User)
    public function chefProjet()
    {
        return $this->belongsTo(User::class, 'chef_projet_id');
    }
        public function client()
{
    return $this->belongsTo(User::class, 'client_id');
}

    // Un projet possède plusieurs tâches
    public function taches()
    {
        return $this->hasMany(Tache::class);
    }

    // Un projet possède plusieurs rapports
    public function rapports()
    {
        return $this->hasMany(Rapport::class);
    }


}